File: Data Structures/dynamicArray.py
#
# Complete the 'dynamicArray' function below.
#
# The function is expected to return an INTEGER_ARRAY.
# The function accepts following parameters:
#  1. INTEGER n
#  2. 2D_INTEGER_ARRAY queries
#

def dynamicArray(n, queries):
    # Write your code here
    print(n, "\n", queries)
    # intialize 2d array named as arr
    # initialize var with named lastAnswer
    # take a loop till the n[1]- (5) times
    # --if the first element of queries is 1 implement q1
    # ----arr(arr[loop time][2nd elment of queries 1d]^lastanswer)%n).append([last element if queries])
    # --if first element of queries[i][j]==2
    # ----idx=arr[loop time][2nd elment of queries 1d]^lastanswer)%n
    # ----assign the value of arr[idx][y%size(arr[idx])] to lastAswaer
    # ----print lastAnswer


dynamicArray([2, 5], [[1, 0, 5], [1, 1, 7],
                      [1, 0, 3],
                      [2, 1, 0],
                      [2, 1, 1]])

File: Data Structures/dynamicArray.js
/*
 * Complete the 'dynamicArray' function below.
 *
 * The function is expected to return an INTEGER_ARRAY.
 * The function accepts following parameters:
 *  1. INTEGER n
 *  2. 2D_INTEGER_ARRAY queries
 */

function dynamicArray(n, queries) {
    // Write your code here
    
    //1- intialize 2d array of n empty arrays
    let arr = [...Array(n)].map(e => Array(0));

    //2- let lastAnswer=0
    let lastAnswer=0

    let ansArray=[]
    let idx=0
    //3- splitting the queries in lines
    var lines = queries.split('\n');
    // console.log(lines)
    //4- loop to the length of string queries
    for ( string_queries in lines)
    {
        //5- splitting the lines in number of array
        var NQ = lines[string_queries].split(' ').map(Number);
        // console.log(NQ)

        let idx=0
        x=NQ[1]

        // console.log("last Answer :",lastAnswer,"\narr[0]",arr[0],'\narr[1]',arr[1])
        //6- queries
        //6.1 - if NQ[0]===1 ?  let idx=(((x && !lastAnswer) || (!x && lastAnswer))%n ) arr[idx]=NQ[2]
        if(NQ[0]===1)
        {
            
            idx=(((x ^ lastAnswer))%n )
            arr[idx].push(NQ[2])
            // console.log('idx',idx,'lastAnswer',lastAnswer,'arr[',idx,'] ',arr[idx])
        }
        // 6.2 - if NQ[0]===2 ? let idx=(((x && !lastAnswer) || (!x && lastAnswer))%n ); lastAnswer=arr[idx][arr[2]%arr[idx].length]
        else if (NQ[0]===2)
        {
            idx=(((x ^ lastAnswer))%n )
            // console.log('index',arr[idx][NQ[2]%arr[idx].length])
            lastAnswer=arr[idx][NQ[2]%arr[idx].length]
            console.log(lastAnswer)
            // console.log('idx',idx,'lastAnswer',lastAnswer,'size(arr[idx])',arr[idx].length)
        }
    }
    
}

dynamicArray(2,"1 0 5\n1 1 7\n1 0 3\n2 1 0\n2 1 1")